internal/nix/input.go

```diff
@@ -45,136 +44,141 @@ type Input struct {
+		if pkgURL.Fragment != "" {
+			normalizedURL += "#" + pkgURL.Fragment
 		}
-		inputURL, _ = url.Parse(normalizedURL)
+		pkgURL, _ = url.Parse(normalizedURL)
 	}
-	return &Input{*inputURL, locker, raw, ""}
+	return &Package{*pkgURL, locker, raw, ""}
 }
 
-// InputFromProfileItem sets the raw Input as the `item`'s unlockedReference i.e.
-// the flake reference and output attribute path used at install time.
-func InputFromProfileItem(item *NixProfileListItem, locker lock.Locker) *Input {
-	return InputFromString(item.unlockedReference, locker)
+// PackageFromProfileItem constructs a package using the the unlocked reference
+// from profile list item.
+func PackageFromProfileItem(item *NixProfileListItem, locker lock.Locker) *Package {
+	return PackageFromString(item.unlockedReference, locker)
 }
 
-// isLocal specifies whether this input is a local flake.
+// isLocal specifies whether this package is a local flake.
 // Usually, this is of the form: `path:./local_flake_subdir#myPackage`
-func (i *Input) isLocal() bool {
+func (p *Package) isLocal() bool {
 	// Technically flakes allows omitting the scheme for local absolute paths, but
 	// we don't support that (yet).
-	return i.Scheme == "path"
+	return p.Scheme == "path"
 }
 
-// isDevboxPackage specifies whether this input is a `canonicalName@version` nix
-// package defined in a devbox.json config. This is in contrast to a "nix" package
-// that can also be a flake or a legacy attribute path.
-func (i *Input) isDevboxPackage() bool {
-	return i.Scheme == ""
+// isDevboxPackage specifies whether this package is a devbox package. Devbox
+// packages have the format `canonicalName@version`and can be resolved by devbox
+// search. This also returns true for legacy packages which are just an
+// attribute path. An explicit flake reference is _not_ a devbox package.
+func (p *Package) isDevboxPackage() bool {
+	return p.Scheme == ""
 }
 
-// isGithub specifies whether this input is a remote flake hosted on a github repository.
+// isGithub specifies whether this Package is referenced by a remote flake
+// hosted on a github repository.
 // example: github:nixos/nixpkgs/5233fd2ba76a3accb5aaa999c00509a11fd0793c#hello
-func (i *Input) isGithub() bool {
-	return i.Scheme == "github"
+func (p *Package) isGithub() bool {
+	return p.Scheme == "github"
 }
 
 var inputNameRegex = regexp.MustCompile("[^a-zA-Z0-9-]+")
 
-// FlakeInputName refers to the name of this Input that will be used in the
-// generated flake.nix. It is unique, and a slug is appended to avoid collisions.
-//
+// FlakeInputName generates a name for the input that will be used in the
+// generated flake.nix to import this package. This name must be unique in that
+// flake
 // Note, that input name has nothing to do with the package name or flake name
 // that it may be referencing. That is the Input.raw field.
-func (i *Input) FlakeInputName() string {
+func (p *Package) FlakeInputName() string {
```

Comment:
line before needs to change `Input.raw` to `Package.raw` 

Or rather, lines 113-114 can be deleted IMO